src/main/groovy/com/github/benmanes/gradle/versions/reporter/HtmlReporter.groovy

```diff
@@ -0,0 +1,287 @@
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.NIGHTLY
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.NIGHTLY
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.RELEASE_CANDIDATE
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.RELEASE_CANDIDATE
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.RELEASE_CANDIDATE
+
+/**
+ * A xml reporter for the dependency updates results.
+ */
+@CompileStatic
+@TupleConstructor(callSuper = true, includeSuperProperties = true, includeSuperFields = true)
+class HtmlReporter extends AbstractReporter {
+
+  String header = """
+    <!DOCTYPE html>
+    <HEAD><TITLE>Project Dependency Updates Report</TITLE></HEAD>
+    <style type=\"text/css\">
+       .body {
+        font:100% verdana, arial, sans-serif;
+        background-color:#fff
+        }
+       .currentInfo {
+           border-collapse: collapse;
+       }
+       .currentInfo td {
+           border: 1px solid black;
+           padding: 12px 15px;
+       }
+       .currentInfo tr:nth-child(even) {
+           background-color: #E4FFB7;
+           padding: 12px 15px;
+       }
+       .currentInfo tr:nth-child(odd) {
+            background-color: #EFFFD2;
+           padding: 12px 15px;
+       }
+
+       .warningInfo {
+           border-collapse: collapse;
+       }
+       .warningInfo td {
+           border: 1px solid black;
+           padding: 12px 15px;
+       }
+       .warningInfo tr:nth-child(even) {
+           background-color: #FFFF66;
+           padding: 12px 15px;
+       }
+       .warningInfo tr:nth-child(odd) {
+            background-color: #FFFFCC;
+           padding: 12px 15px;
+       }
+   </style>"""
+
+  @Override
+  def write(printStream, Result result) {
+    writeHeader(printStream)
+
+    if (result.count == 0) {
+      printStream.println '<P>No dependencies found.</P>'
+    } else {
+      writeUpToDate(printStream, result)
+      writeExceedLatestFound(printStream, result)
+      writeUpgrades(printStream, result)
+      writeUnresolved(printStream, result)
+    }
+
+    writeGradleUpdates(printStream, result)
+  }
+
+  private def writeHeader(printStream) {
+    printStream.println header.stripMargin()
+  }
+
+  private def writeUpToDate(printStream, Result result) {
+    SortedSet<Dependency> versions = result.getCurrent().getDependencies()
+    if (!versions.isEmpty()) {
+      printStream.println("<H2>Current dependencies</H2>")
+      printStream.println("<p>The following dependencies are using the latest ${revision} version:<p>")
+      printStream.println("<table class=\"currentInfo\">")
+      getCurrentRows(result).each { printStream.println it }
+      printStream.println("</table>")
+      printStream.println("<br>")
+    }
+  }
+
+  private def getCurrentRows(Result result) {
```

Comment:
You might want some type of show/hide button or collapsible header, with current being hidden by default. Most don't really care about what is up to date. In the console report it's easy to ignore since your terminal is scrolling bottom to top.

I decided to show it mostly to debug with, e.g. see if any dependencies were missing or incorrectly categorized. I think it's better to have than not, but is also the least interesting part of the report.

Reply:
Interesting thought. I can look into that.  I maintained the order from the text report, but your use case makes sense.

Reply:
next commit.